src/CommunityToolkit.Maui.Core/Extensions/UIViewExtensions.macios.cs

```diff
@@ -84,4 +84,30 @@ public static class UIViewExtensions
 		UIDevice.CurrentDevice.CheckSystemVersion(11, 0)
 			? view.SafeAreaLayoutGuide.WidthAnchor
 			: view.WidthAnchor;
+
+	/// <summary>
+	/// Get Key Window
+	/// </summary>
+	public static UIWindow? GetKeyWindow()
+	{
+		return UIApplication.SharedApplication.ConnectedScenes.ToArray()
+			.Select(x => x as UIWindowScene)
+			.FirstOrDefault()?
+			.Windows.FirstOrDefault(x => x.IsKeyWindow);
+	}
+	
+	/// <summary>
+	/// Get current UI view controller
+	/// </summary>
+	public static UIViewController? GetCurrentUIViewController()
```

Comment:
why not use the `Microsoft.Maui.ApplicationModel.GetCurrentUIViewController` method?